core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/HermitePolynomial.java

```diff
@@ -0,0 +1,106 @@
+import static java.lang.Math.pow;
+import static show.ezkz.casprzak.core.tools.MiscTools.factorial;
+import static show.ezkz.casprzak.core.tools.defaults.DefaultFunctions.*;
+
+/**
+ * The methods in {@link HermitePolynomial} deal with <a href="https://en.wikipedia.org/wiki/Hermite_polynomials#Hermite_functions">Hermite Polynomials</a>.
+ */
+public class HermitePolynomial {
+
+	private static final Map<Integer, GeneralFunction> cache = new HashMap<>();
+
+	/**
+	 * Returns the nth Hermite polynomial with the default variable defined with {@link Settings#singleVariableDefault}
+	 * @param n the nth polynomial
+	 * @return nth Hermite polynomial
+	 */
+	public static GeneralFunction hermitePolynomial(int n) {
+		if (Settings.cacheDerivatives && cache.containsKey(n))
+			return cache.get(n);
+
+		GeneralFunction[] sum = new GeneralFunction[(int) floor(n / 2.) + 1];
+		for (int m = 0; m <= (int) Math.floor(n / 2.); m++) {
+			sum[m] = new Product(new Constant(constant(n, m)), new Pow(new Constant(n - 2 * m), new Variable(Settings.singleVariableDefault))).simplify();
+		}
+		GeneralFunction polynomial = new Sum(sum).simplify();
+		cache.put(n, polynomial);
+		return polynomial;
+	}
+
+	/**
+	 * Returns the nth Hermite polynomial with a variable of the specified String
+	 * @param n the nth polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return nth Hermite polynomial
+	 */
+	public static GeneralFunction hermitePolynomial(int n, String variable) {
+		GeneralFunction polynomial = hermitePolynomial(n);
+		if (Settings.singleVariableDefault.equals(variable))
+			return polynomial;
+		else {
+			Map<String, Variable> substitution = new HashMap<>() {
+				{
+					put(Settings.singleVariableDefault, new Variable(variable));
+				}
+			};
+			return polynomial.substituteVariables(substitution);
+		}
+	}
+
+	/**
+	 * Returns the nth normalized Hermite polynomial with the default variable defined with {@link Settings#singleVariableDefault}
+	 * @param n the nth polynomial
+	 * @return nth normalized Hermite polynomial
+	 */
+	public static GeneralFunction normalHermitePolynomial(int n) {
+		return new Product(reciprocal(normalizingConstant(n)), hermitePolynomial(n)).simplify();
+	}
+
+	/**
+	 * Returns the nth normalized Hermite polynomial with a variable of the specified String
+	 * @param n the nth polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return nth normalized Hermite polynomial
+	 */
+	public static GeneralFunction normalHermitePolynomial(int n, String variable) {
+		return new Product(reciprocal(normalizingConstant(n)), hermitePolynomial(n, variable)).simplify();
+	}
+
+	/**
+	 * The normalizing constant of the nth Hermite polynomial
+	 * ie If you wanted to normalize, you would do {@code 1/N * L_n}
+	 * @param n the nth polynomial
+	 * @return normalizing constant of the nth Hermite polynomial
+	 */
+	public static GeneralFunction normalizingConstant(int n) {
```

Comment:
Shouldn't this return 1/N instead? That'd make the other methods simpler too

Reply:
No.